src/pools/D3MPoolBase.sol

```diff
@@ -70,26 +114,90 @@ abstract contract D3MPoolBase {
         CanLike(d3mHubLike(hub_).vat()).hope(hub_);
     }
 
+    /**
+       @notice Logic for the specific pool that determines whether it is valid.
+       @dev This is used in the D3M Hub to allow permissionless caging of the pool.
+       @return bool true or false
+    */
     function validTarget() external view virtual returns (bool);
 
+    /**
+        @notice Function that will deposit assets (Dai) in the external pool.
+        @dev If the external pool requires a different amount be passed in the
+        conversion should occur here as the Hub passes Dai [wad] amounts.
+        @param amt Amount in Dai/asset terms that we want to deposit
+    */
     function deposit(uint256 amt) external virtual;
 
+    /**
+        @notice Function that will withdraw assets (Dai) from the external pool.
+        @dev If the external pool requires a different amount be passed in the
+        conversion should occur here as the Hub passes Dai [wad] amounts.
+        @param amt Amount in Dai/asset terms that we want to withdraw
+    */
     function withdraw(uint256 amt) external virtual;
 
+    /**
+        @notice Function that will transfer shares.
+        @dev If the external pool/shares contract requires a different amount be passed in the
+        conversion should occur here as the Hub passes Gem [wad] amounts.
+        @param dst Address that should receive the shares
+        @param amt Amount in Gem terms that we want to withdraw
+        @return bool True|False of whether the transfer was successful per ERC-20 standard
+    */
     function transfer(address dst, uint256 amt) external virtual returns (bool);
 
+    /**
+        @notice Function that will transfer all shares from this pool
+        @param dst Address that should receive the shares
+        @return bool True|False of whether the transfer was successful per ERC-20 standard
+    */
     function transferAll(address dst) external virtual returns (bool);
 
+    /**
+        @notice Some external pools require manually accruing fees/interest
+        @dev Used in Hub.exec() and Hub.reap(), can be empty if external pool does not require
+    */
     function accrueIfNeeded() external virtual;
 
+    /**
+        @notice Balance of Assets this pool "owns".
+        Should be Deposit + Interest/Fees earned
```

Comment:
In general it will not be specifically just deposit + interest. It could subtract out losses as well. I would just leave it open instead of defining what this should equal.